src/storage/txn/commands/prewrite_pessimistic.rs

```diff
@@ -77,21 +80,37 @@ impl<S: Snapshot, L: LockManager, P: PdClient + 'static> WriteCommand<S, L, P>
         // for getting the written keys.
         txn.extra_op = context.extra_op;
 
+        let primary_key = Key::from_raw(&self.primary);
         let mut locks = vec![];
+        let mut async_commit_ts = TimeStamp::zero();
         for (m, is_pessimistic_lock) in self.mutations.clone().into_iter() {
+            let mut secondaries = &self.secondary_keys.as_ref().map(|_| vec![]);
+
+            if m.key() == &primary_key {
```

Comment:
How about
```suggestion
            if !secondaries.is_empty() && m.key() == &primary_key {
```

Reply:
Sorry I don't understand...
`secondaries` is an `Option<Vec<Vec<u8>>`. It is `Some` when async commit is enabled. So it's `Some(vec![])` if its a secondary key.

Reply:
😅 I'm wrong. Now we have to compare every key with the primary key even if the transaction isn't an async-commit one. We can eliminate it in this ugly manner.
```
            if self.secondary_keys.as_ref().map(|s| !s.is_empty()).unwrap_or_default() && m.key() == &primary_key {
```

Reply:
makes sense

Reply:
But i think we only need to check if `secondaries` is some

Reply:
In my manner, only the batch containing the primary key has to do the comparison.

Reply:
i see what you mean

Reply:
I try to optimize it, ptal